buildscripts/Set-VersionInfo.ps1

```diff
@@ -0,0 +1,58 @@
+<#
+
+.SYNOPSIS
+
+Extracts build version information from Settings.proj and the
+environment. Then constructs several flavor version strings and
+uses them, along with the current year to
+
+  - update buildscripts/CommonAssemblyInfo.cs,
+  - update all project.json files, and
+  - emit TeamCity directives so it can know about the values
+
+#>
+
+[xml] $settings = Get-Content 'Settings.proj'
```

Comment:
I was thinking of moving this into TeamCity in the "Build number format", but happy for it to continue using Settings.proj.